Competition

AknA13 opened the pull request.

Commit:
added all swerve drive code

Commit:
checkstyle

Commit:
checkstyle

Commit:
checkstyle

Commit:
added changes needed for merge

Commit:
added changes needed for merge 2

Commit:
changed spelling

Commit:
changed can spark order

AknA13 committed:
added updated mech code

Commit:
pid

Commit:
Merge branch 'competition' of https://github.com/Tino-FRC-2473/FRC2023-Offseason into competition

Commit:
uncommented robot.java and reset pid

Commit:
testing mech

AknA13 committed:
added back homing

AknA13 committed:
elevator encoder saves between re-enables

AknA13 committed:
got rid of unused auto methods

AknA13 committed:
fixed encoders for arm

AknA13 committed:
changed elevator powers, added zeroing back for wrist

AknA13 committed:
Pushed wrist testing changes

AknA13 committed:
added previous encoder bounds for wrist

AknA13 committed:
Added velocity PID code

AknA13 committed:
Merge branch 'main' into competition

AknA13 committed:
removed unused code

beranki committed:
tweaking encoder vals + testing velocity pid

Commit:
10/3 drive testing changes

AknA13 committed:
maintains flip orientation in idle

AknA13 committed:
encoder tweaks

Commit:
changed throttle curve to squared

Commit:
Merge branch 'competition' of https://github.com/Tino-FRC-2473/FRC2023-Offseason into competition

Commit:
changed throttle curve to squared

AknA13 committed:
testing code

Commit:
Added auto outtaking

Commit:
resolved merge

Commit:
10/5 code changes

AknA13 committed:
tweaked intake code for state transitions

Commit:
auto dist changes

AknA13 committed:
removed unused code, tweaked flip encoder margins

Commit:
increased wrist speed to 0.15 and auto speed

Commit:
Merge branch 'competition' of https://github.com/Tino-FRC-2473/FRC2023-Offseason into competition

Commit:
added wrist states for double power movement in case of CAn connection being weak

beranki committed:
fixed issue with the wrist double state

AknA13 committed:
added webcame capture code

beranki committed:
tracking auto

AknA13 committed:
changes for mech deposit auto

AknA13 committed:
fixed checkstyle

AknA13 committed:
updated auto outtaking state handle method

AknA13 committed:
uncommented auto drive code

AknA13 committed:
Merge branch 'main' into competition

AknA13 committed:
removed duplicate PID

tsoim commented on src/main/java/frc/robot/SwerveConstants.java:
```diff
@@ -126,15 +125,17 @@ public static final class OIConstants {
+		public static final double MAX_SPEED_METERS_PER_SECOND = 2; //3
+		public static final double MAX_ACCELERATION_METERS_PER_SECOND_SQUARED = 2; //3
 		public static final double MAX_ANGULAR_SPEED_RADIANS_PER_SECOND = Math.PI / 2; // pi
 		public static final double MAX_ANGULAR_SPEED_RADIANS_PER_SECOND_SQUARED = Math.PI / 2; // pi
```
the comments after each variable declaration is stale, please update

tsoim commented on src/main/java/frc/robot/TeleopInputXBox.java:
this whole file is just commented out? please clean

tsoim commented on src/main/java/frc/robot/systems/ElevatorWristFSM.java:
```diff
@@ -19,17 +23,22 @@ public class ElevatorWristFSM {
+	private static final float MAX_UP_POWER = -0.15f;
+	private static final float MAX_DOWN_POWER = 0.15f;
 	private static final double WRIST_IN_ENCODER_ROTATIONS = 200; //16
 	private static final double WRIST_OUT_ENCODER_ROTATIONS = -200; //-40
```
stale comments

tsoim commented on src/main/java/frc/robot/systems/SpinningIntakeFSM.java:
why is this removed? are we only using EveryBotIntakeFSM?

AknA13 replied:
yes, SpinningIntakeFSM was the FSM for last year's intake

tsoim reviewed (commented):
can you add a comment of what is enabled/disabled as part of this merge?

AknA13 committed:
Cleaned up commented code

AknA13 commented:
Competition changes:
-Increased autonomous drive distances and speeds
-Switched TeleopInput code from XBox controller to PS4 Controller
-Removed factory resetting code for SparkMaxs
-Fixed bug with intaking while flipped
-Added outtaking state for autonomous
-Enabled webcam feed
-Added double power states for wrist movement
-Added autonomous extension and retraction states for elevator
-Autonomous code has mid deposit and mobility
